web/src/engine/websites/SpoilerPlus.ts

```diff
@@ -0,0 +1,189 @@
+            let difference = 0;
+            const divClasses = document.getElementById('cchapter') ? [...document.getElementById('cchapter').classList]: undefined;
+            const ccChapterRgx = /cchapter-(\\d+)/;
+            if (divClasses) {
+                const goodClass = divClasses.find(element => ccChapterRgx.test(element));
+                if (goodClass) {
+                    difference = parseInt(goodClass.match(ccChapterRgx)[1]);
+                }
+            }
+            const scrambleArray = scrambleSeed.match(/.{1,2}/g).map(a => {
+                return parseInt(a) - difference;
+            });
+            resolve({images : images, scrambleArray : scrambleArray });
+        }
+    });
+
+`;
+
+type PageData = {
+    scrambleArray: number[],
+    images: string[]
+}
+
+type TDimension = {
+    height: number,
+    width: number
+}
+
+type TPiece = {
+    height: number,
+    left: number,
+    top: number,
+    width: number
+}
+
+type TPuzzleData = {
+    from: TPiece,
+    to: TPiece
+}
+
+function MangaLabelExtractor(element: HTMLElement): string {
+    return StripCrap(element.textContent);
+}
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname,
+        title: StripCrap(anchor.title)
+    };
+}
+
+@Common.MangaCSS(/^{origin}\/[^/]+-raw-free\/$/, 'article#item-detail h1.title-detail', MangaLabelExtractor)
+@Common.MangasMultiPageCSS('/page/{page}/', 'article.item div.image > a', 1, 1, 0, MangaInfoExtractor)
+@Common.ChaptersSinglePageCSS('div.list-chapter ul li a')
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('spoilerplus', 'SpoilerPlus', 'https://spoilerplus.tv', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Aggregator, Tags.Accessibility.RegionLocked);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const data = await FetchWindowScript<PageData>(new Request(new URL(chapter.Identifier, this.URI)), pageScript, 500);
+        return data.images.map(image => new Page(this, chapter, new URL(image), { Referer: this.URI.origin, scrambleArray: JSON.stringify(data.scrambleArray) }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal);
+        if (!page.Parameters?.scrambleArray) return blob;
+        const scrambleArray = JSON.parse(page.Parameters.scrambleArray as string) as number[];
+        return DeScramble(blob, async (image, ctx) => {
+            function COMPUTEPIECES(skey: number[], image: ImageBitmap): TPuzzleData[] {
+                let obj1: TDimension = {
+                    width: image.width,
+                    height: image.height,
+                };
+                let obj2: TDimension = {
```

Comment:
It is identical to `obj1`, why creating a duplicate?

Reply:
one on those used image.naturalheight and  image.naturalwidth, which are unavailable in ImageBitmap .
After reading this https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/naturalHeight i think for our use case (we directly fetch the image), height and width would be a reasonable replacement.

then, i missed the dupe it created :D